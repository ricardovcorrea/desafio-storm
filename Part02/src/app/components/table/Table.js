import React from "react";
import "./Table.scss";

import PropTypes from "prop-types";

import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import Checkbox from "@material-ui/core/Checkbox";
import Icon from "@material-ui/core/Icon";
import LinearProgress from "@material-ui/core/LinearProgress";

import moment from "moment";

import Pagination from "../pagination/Pagination";

const CustomTable = props => {
  return (
    <React.Fragment>
      <Paper>
        <TableContainer>
          <Table size={"medium"}>
            <TableHead>
              <TableRow>
                <TableCell padding="checkbox"></TableCell>
                <TableCell padding="checkbox"></TableCell>
                <TableCell>USUÁRIO</TableCell>
                <TableCell>EMAIL</TableCell>
                <TableCell className={"centerText"}>DATA DE INCLUSÃO</TableCell>
                <TableCell className={"centerText"}>
                  DATA DE ALTERAÇÃO
                </TableCell>
                <TableCell className={"centerText"}>REGRAS</TableCell>
                <TableCell className={"centerText"}>STATUS</TableCell>
                <TableCell padding="checkbox"></TableCell>
                <TableCell className={"centerText"}>AÇÕES</TableCell>
              </TableRow>
            </TableHead>

            <TableBody>
              {!props.isLoading &&
                props.users.map((row, index) => {
                  return (
                    <TableRow key={row.name} className={"row"}>
                      <TableCell padding="checkbox">
                        <Checkbox />
                      </TableCell>
                      <TableCell padding="checkbox"></TableCell>
                      <TableCell>{row.name}</TableCell>
                      <TableCell>{row.email}</TableCell>
                      <TableCell className={"centerText"}>
                        {moment(row.createDate).format("DD/MM/YYYY")}
                      </TableCell>
                      <TableCell className={"centerText"}>
                        {moment(row.changeDate).format("DD/MM/YYYY")}
                      </TableCell>
                      <TableCell className={"centerText"}>
                        {row.rules}
                      </TableCell>
                      <TableCell className={"centerText"}>
                        <span
                          style={
                            row.status ? { color: "green" } : { color: "red" }
                          }
                        >
                          {row.status ? "ATIVO" : "INATIVO"}
                        </span>
                      </TableCell>
                      <TableCell padding="checkbox"></TableCell>
                      <TableCell className={"centerText"}>
                        <Icon>more_horiz</Icon>
                      </TableCell>
                    </TableRow>
                  );
                })}
            </TableBody>
          </Table>
          {props.isLoading && <LinearProgress />}
        </TableContainer>
      </Paper>
      {!props.isLoading && <Pagination pages={1} />}
    </React.Fragment>
  );
};

CustomTable.propTypes = {
  users: PropTypes.array.isRequired,
  isLoading: PropTypes.bool
};

export default CustomTable;
