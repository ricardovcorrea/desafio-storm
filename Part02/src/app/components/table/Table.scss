.row:nth-child(odd) {
  background-color: #f5f5f5;
}

.row:nth-child(even) {
  background-color: #e9e9e9;
}

.centerText {
    text-align: center !important;
}