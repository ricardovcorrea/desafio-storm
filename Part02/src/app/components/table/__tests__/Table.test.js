import React from "react";
import { render } from "@testing-library/react";
import Table from "../Table";

test("renders table", () => {
  const container = render(<Table users={[]} />);
  expect(container.container).toMatchSnapshot();
});
