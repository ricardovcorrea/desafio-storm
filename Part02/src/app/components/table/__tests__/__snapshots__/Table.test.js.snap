// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders table 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
              scope="col"
            />
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
              scope="col"
            />
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              USUÁRIO
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head"
              scope="col"
            >
              EMAIL
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head centerText"
              scope="col"
            >
              DATA DE INCLUSÃO
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head centerText"
              scope="col"
            >
              DATA DE ALTERAÇÃO
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head centerText"
              scope="col"
            >
              REGRAS
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head centerText"
              scope="col"
            >
              STATUS
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-paddingCheckbox"
              scope="col"
            />
            <th
              class="MuiTableCell-root MuiTableCell-head centerText"
              scope="col"
            >
              AÇÕES
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root row"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-57 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-60"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
            />
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              teste01
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              teste01@gmail.com
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body centerText"
            >
              31/07/2019
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body centerText"
            >
              05/03/2020
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body centerText"
            >
              01
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body centerText"
            >
              <span
                style="color: red;"
              >
                INATIVO
              </span>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
            />
            <td
              class="MuiTableCell-root MuiTableCell-body centerText"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                more_horiz
              </span>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root row"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
            >
              <span
                aria-disabled="false"
                class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-57 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <input
                    class="PrivateSwitchBase-input-60"
                    data-indeterminate="false"
                    type="checkbox"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </span>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
            />
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              teste02
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body"
            >
              teste02@gmail.com
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body centerText"
            >
              08/02/2019
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body centerText"
            >
              03/05/2020
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body centerText"
            >
              02
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body centerText"
            >
              <span
                style="color: green;"
              >
                ATIVO
              </span>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-paddingCheckbox"
            />
            <td
              class="MuiTableCell-root MuiTableCell-body centerText"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                more_horiz
              </span>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
  <div
    class="pagination"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained paginationButtonInactive MuiButton-disableElevation"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Primeiro
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained paginationButtonInactive MuiButton-disableElevation"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Anterior
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained button"
      style="background-color: rgb(216, 51, 103);"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        <span
          style="color: white;"
        >
          1
        </span>
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained paginationButtonInactive MuiButton-disableElevation"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Próximo
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained paginationButtonInactive MuiButton-disableElevation"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        Último
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
